registration/registration.go

```diff
@@ -195,8 +195,11 @@ func (r *Registration) Run(ctx context.Context) error {
 	proofs := r.workerSvc.RegisterForProofs(ctx)
 
 	// First re-execute the in-progress round if any
-	if err := r.recoverExecution(ctx); err != nil {
-		return fmt.Errorf("recovering execution: %w", err)
+	if r.openRound.epoch > 0 {
+		epoch := uint(int(r.openRound.epoch) - 1)
+		if err := r.recoverExecution(ctx, epoch); err != nil {
+			return fmt.Errorf("recovering execution: %w", err)
+		}
 	}
```

Comment:
You are already checking that `epoch > 0` so the cast shouldn't be necessary. Even if `epoch` would be 0 with or without the cast the result would be `math.MaxUint`

```suggestion
	if r.openRound.epoch > 0 {
		if err := r.recoverExecution(ctx, r.openRound.epoch-1); err != nil {
			return fmt.Errorf("recovering execution: %w", err)
		}
	}
```